### PR TITLE
Intruduction cucumber fwk

### PR DESCRIPTION
**Pull request by volna80**
Hi,

I am writing an articule about using Cucumber and select your project randomly. 

The goal was just to show how it is possible to use Cucumber for writing integration tests and how they add a bit more readability to integration test scenarios. 

If you like the result, feel free to accept my PR.

I am fine if you decide not adding one more technology to the project and keep it a bit simpler. 

If you are Russian speaker, I can send you a link when I pubilsh my article.

### CODE REVIEW

**Comment by mzheravin** [comment]
Hi! Thats looks interesting. Please give me some time to check if it can cause issues and not very difficult to maintain.

File: src/test/java/exchange/core2/tests/util/ExchangeTestContainer.java [hunk 1]
```diff
@@ -372,11 +372,12 @@ public void submitMultiCommandSync(ApiCommand dataCommand) {
     void submitCommandsSync(List<ApiCommand> apiCommand) throws InterruptedException {
         final CountDownLatch latch = new CountDownLatch(apiCommand.size());
         consumer = cmd -> {
-            assertEquals(CommandResultCode.SUCCESS, cmd.resultCode);
-            latch.countDown();
+            if (cmd.resultCode == CommandResultCode.SUCCESS) {
```
[the remark this answers is not in the document]
Comment (volna80) [reply src/test/java/exchange/core2/tests/util/ExchangeTestContainer.java]: Ok

File: src/test/java/exchange/core2/tests/steps/OrderBookStepdefs.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+public class OrderBookStepdefs {
+
+    private final ExchangeTestContainer container;
+
+    public OrderBookStepdefs(ExchangeTestContainer container) {
+        this.container = container;
+    }
+
+    @Then("An {symbol} order book is:")
+    public void an_order_book_is(CoreSymbolSpecification symbol, List<List<String>> dataTable) {
```
[the remark this answers is not in the document]
Comment (volna80) [reply src/test/java/exchange/core2/tests/steps/OrderBookStepdefs.java]: Sure,  acutally I decided to miss it in purpose of to simplify my article. 

If you like the idea, I will try to improve this refactoring and probably migrate a couple more JUnit tests to cucumber.

[the remark this answers is not in the document]
Comment (mzheravin) [reply src/test/java/exchange/core2/tests/steps/OrderBookStepdefs.java]: Contribution is very appreciated :+1:  -  I will merge it as soon as it has no conflicts (mostly regarding to my first comment above). TableTransformer is up to you.

File: src/test/java/exchange/core2/tests/util/ExchangeTestContainer.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (mzheravin) [reply src/test/java/exchange/core2/tests/util/ExchangeTestContainer.java]: I found out that solution I proposed above does not work well with recent master branch. 
Perfrmance and integration test are not stable if extra commands with ACCEPTED responses are not ignored for counter. I will investigate asap. So far you can use implementation from master:
```
        consumer = cmd -> {
            assertTrue(CommandResultCode.SUCCESS == cmd.resultCode || CommandResultCode.ACCEPTED == cmd.resultCode);
            latch.countDown();
        };
```
Feel free to add your own implementation of `submitCommandsSync` if you need different semantics. These methods design needs to be revised anyway.

**Comment by mzheravin** [comment]
Looks good. 
Thanks so much for your contribution!
Merging...

**Pull request merged**